Docker/Component.php

```diff
@@ -0,0 +1,211 @@
+<?php
+
+namespace Keboola\DockerBundle\Docker;
+
+use Keboola\Gelf\ServerFactory;
+use Keboola\Syrup\Exception\ApplicationException;
+use Symfony\Component\Config\Definition\Exception\InvalidConfigurationException;
+
+class Component
+{
+    /**
+     * @var string
+     */
+    private $id;
+
+    /**
+     * @var array
+     */
+    private $data;
+
+    /**
+     * @var string
+     */
+    private $networkType;
+
+    /**
+     * Component constructor.
+     * @param array $componentData Component data as returned by Storage API
+     */
+    public function __construct(array $componentData)
+    {
+        $this->id = empty($componentData['id']) ? '' : $componentData['id'];
+        $data = empty($componentData['data']) ? [] : $componentData['data'];
+        try {
+            $this->data = (new Configuration\Component())->parse(['config' => $data]);
+        } catch (InvalidConfigurationException $e) {
+            throw new ApplicationException("Image definition is empty or invalid: " . $e->getMessage(), $e, $data);
+        }
+        $this->networkType = $this->data['network'];
+    }
+
+    /**
+     * @return string
+     */
+    public function getId()
+    {
+        return $this->id;
+    }
+
+    /**
+     * @return string
+     */
+    private function getSanitizedComponentId()
+    {
+        return preg_replace('/[^a-zA-Z0-9-]/i', '-', $this->getId());
+    }
+
+    /**
+     * @return string
+     */
+    public function getConfigurationFormat()
```

Comment:
Nemyslel jsem, že `$this->data` jsou defaulty, defaulty se do toho doplní z ` $this->data = (new Configuration\Component())->parse(['config' => $data]);`